File: benchmark/results/ts-multilang/run-005/b-baseline/output/src/domain/inventory/services/ReservationService.ts
import { StockItem } from "../aggregates/StockItem";

/** Service for managing stock reservations. */
export class ReservationService {
  create(input: Partial<StockItem>): StockItem {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): StockItem | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): StockItem[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}

File: benchmark/results/python-multilang/run-004/b-baseline/output/metaengine_demo/inventory/services/reservation_service.py
"""ReservationService for the inventory domain."""
from typing import Optional

from metaengine_demo.inventory.aggregates.stock_item import StockItem


class ReservationService:
    """Service managing stock reservations."""

    def create(self, input: Optional[StockItem]) -> StockItem:
        """Create a reservation for a StockItem."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[StockItem]:
        """Find a reserved StockItem by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[StockItem]:
        """List reserved StockItems up to limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Cancel a reservation for a StockItem by id."""
        raise NotImplementedError("not implemented")
